openforcefield/typing/engines/smirnoff/parameters.py

```diff
@@ -652,6 +646,40 @@ def __init__(self, permit_cosmetic_attributes=False, **kwargs):
                                         "'permit_cosmetic_attributes=True'".format(key, self.__class__))
 
 
+    def _add_parameters(self, section_dict, permit_cosmetic_attributes=False):
```

Comment:
I've separated the parameter addition code out from the ParameterHandler initialization code. This way, we can go through multiple rounds of parameter addition from different SMIRNOFF data sources with a single ParameterHandler.

Reply:
Good call!